### PR TITLE
chore: upgrade chevrotain to v9.0.2

### LINKED ISSUE #423: Upgrade to chevrotain 7
**Why ?**

We should try to have latest versions of our libs to include easily their improvements (bugs, enhancements, etc.)

**What ?**

I tried to upgrade, but ran into an error:

```bash

jhipster/prettier-java/node_modules/chevrotain/lib/src/parse/parser/traits/recognizer_engine.js:184
            throw e;
            ^

TypeError: Cannot read property 'location' of undefined
    at JavaParser.TreeBuilder.cstPostNonTerminal (/Users/clementdessoude/Documents/dev/jhipster/prettier-java/node_modules/chevrotain/lib/src/parse/parser/traits/tree_builder.js:160:73)
    at JavaParser.cstPostNonTerminal (/Users/clementdessoude/Documents/dev/jhipster/prettier-java/packages/java-parser/src/parser.js:81:11)
    at JavaParser.RecognizerEngine.subruleInternal (/Users/clementdessoude/Documents/dev/jhipster/prettier-java/node_modules/chevrotain/lib/src/parse/parser/traits/recognizer_engine.js:423:18)
    at JavaParser.RecognizerApi.SUBRULE (/Users/clementdessoude/Documents/dev/jhipster/prettier-java/node_modules/chevrotain/lib/src/parse/parser/traits/recognizer_api.js:72:21)
    at JavaParser.<anonymous> (/Users/clementdessoude/Documents/dev/jhipster/prettier-java/packages/java-parser/src/productions/interfaces.js:241:7)
    at JavaParser.invokeRuleWithTry (/Users/clementdessoude/Documents/dev/jhipster/prettier-java/node_modules/chevrotain/lib/src/parse/parser/traits/recognizer_engine.js:117:33)
    at JavaParser.wrappedGrammarRule [as annotation] (/Users/clementdessoude/Documents/dev/jhipster/prettier-java/node_modules/chevrotain/lib/src/parse/parser/traits/recognizer_engine.js:131:38)
    at JavaParser.RecognizerEngine.doSingleRepetition (/Users/clementdessoude/Documents/dev/jhipster/prettier-java/node_modules/chevrotain/lib/src/parse/parser/traits/recognizer_engine.js:385:16)
    at JavaParser.RecognizerEngine.manyInternalLogic (/Users/clementdessoude/Documents/dev/jhipster/prettier-java/node_modules/chevrotain/lib/src/parse/parser/traits/recognizer_engine.js:318:29)
    at JavaParser.RecognizerEngine.manyInternal (/Users/clementdessoude/Documents/dev/jhipster/prettier-java/node_modules/chevrotain/lib/src/parse/parser/traits/recognizer_engine.js:295:21)
```

### PR DESCRIPTION
**Pull request by clementdessoude**
fix #423

### CODE REVIEW

File: packages/java-parser/src/productions/expressions.js [hunk 1]
```diff
@@ -309,12 +309,7 @@ function defineRules($, t) {
       $.SUBRULE($.annotation);
     });
 
-    $.OPTION({
-      NAME: "$methodTypeArguments",
-      DEF: () => {
-        $.SUBRULE2($.typeArguments);
-      }
-    });
+    $.OPTION(() => $.SUBRULE2($.typeArguments));
```
Comment (bd82) [inline packages/java-parser/src/productions/expressions.js:312]: this change is not related to the upgrade correct?

Comment (clementdessoude) [reply packages/java-parser/src/productions/expressions.js:312]: Support for nested In line rules has been deprecated in v7.0.0, if I understand the changelog correctly: https://chevrotain.io/docs/changes/BREAKING_CHANGES.html#_7-0-0

Comment (bd82) [reply packages/java-parser/src/productions/expressions.js:312]: okay

**Comment by DanielFran** [comment]
Superseded by https://github.com/jhipster/prettier-java/pull/616
